### ISSUE TITLE
Node templates not autogenerated

### ISSUE THREAD

**Issue by klfman** [Flowpack/Flowpack.NodeTemplates#30]
(Might be outdated, as I had no time to reproduce this with the newest version. My Version was 1.1.1. **Edit**: I tested it with 1.3.1 and the issue still exists).

It seems to me that the node template config does not work (does not insert nodes automatically) if its parent is a configured childNode, meaning, the user did not insert it itself but it was generated as configured under the yaml key childNodes (of some higher parent in the hierarchy).

To put it another way: To me it looks like there is some hook that is not called when nodes are inserted by "the framework" instead of the user.

**Comment by daniellienert** [comment]
Hey @klfman,

that is correct. The nodeTemplate is triggered as a NodeCreationHandler directly when inserting it via the UI, not when a node is inserted by some code.

**Comment by klfman** [comment]
@daniellienert So do you think that's even fixable? Sounds like a different strategy would be needed to generate these nodes then

**Comment by theilm** [comment]
Hi @klfman, actually it's part of the concept; node templates are only triggered by the node that is inserted via UI. I do not know your actual use case, but you can create nested structures in the node template - so you can modify your childNode (or create childNodes for it) by creating a node template for the parent node that is actually inserted.

**Comment by klfman** [comment]
It's not a bug, it's a feature? :D 
Okay, I'll find a way around with the nested structures in the parent, thank you all!
